client/components/backup-retention-management/index.tsx

```diff
@@ -70,53 +73,104 @@ const BackupRetentionManagement: FunctionComponent = () => {
 	const storageLimitBytes = useSelector( ( state ) =>
 		getRewindBytesAvailable( state, siteId )
 	) as number;
-	const estimatedCurrentSiteSize = useEstimatedCurrentSiteSize();
 
+	const lastBackupSize = useSelector( ( state ) =>
+		getBackupCurrentSiteSize( state, siteId )
+	) as number;
+
+	const estimatedCurrentSiteSize = getEstimatedCurrentSiteSize( lastBackupSize );
 	const currentSiteSizeText = useStorageText( estimatedCurrentSiteSize );
 	const storageLimitText = useStorageText( storageLimitBytes );
 
-	const retentionOptionsCards: Record< number, RetentionRadioOptionType > = {
-		[ RETENTION_OPTIONS.RETENTION_DAYS_7 ]: usePrepareRetentionOptions(
-			translate( '7 days' ),
-			RETENTION_OPTIONS.RETENTION_DAYS_7,
-			currentRetentionPlan,
-			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_7
-		),
-		[ RETENTION_OPTIONS.RETENTION_DAYS_30 ]: usePrepareRetentionOptions(
-			translate( '30 days' ),
-			RETENTION_OPTIONS.RETENTION_DAYS_30,
-			currentRetentionPlan,
-			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_30
-		),
-		[ RETENTION_OPTIONS.RETENTION_DAYS_120 ]: usePrepareRetentionOptions(
-			translate( '120 days' ),
-			RETENTION_OPTIONS.RETENTION_DAYS_120,
-			currentRetentionPlan,
-			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_120
-		),
-		[ RETENTION_OPTIONS.RETENTION_DAYS_365 ]: usePrepareRetentionOptions(
-			translate( '1 year' ),
-			RETENTION_OPTIONS.RETENTION_DAYS_365,
-			currentRetentionPlan,
-			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_365
-		),
-	};
+	const [ retentionOptionsCards, setRetentionOptionsCards ] = useState<
+		RetentionRadioOptionType[]
+	>( [
+		{
+			label: translate( '7 days' ),
+			spaceNeededInBytes: 0,
+			upgradeRequired: false,
+			isCurrentPlan: false,
+			value: 7,
+			checked: false,
+		},
+		{
+			label: translate( '30 days' ),
+			spaceNeededInBytes: 0,
+			upgradeRequired: false,
+			isCurrentPlan: false,
+			value: 30,
+			checked: false,
+		},
+		{
+			label: translate( '120 days' ),
+			spaceNeededInBytes: 0,
+			upgradeRequired: false,
+			isCurrentPlan: false,
+			value: 120,
+			checked: false,
+		},
+		{
+			label: translate( '1 year' ),
+			spaceNeededInBytes: 0,
+			upgradeRequired: false,
+			isCurrentPlan: false,
+			value: 365,
+			checked: false,
+		},
+	] );
```

Comment:
This object will be reinitialized on every re-render. We could benefit from use lazy initialization. See [this](https://kentcdodds.com/blog/use-state-lazy-initialization-and-function-updates) for more info.

It also feels like a weird data type to live in state. Let's try to refactor and use as little state as possible. The rest that is always the same can live outside of the state.

Reply:
> This object will be reinitialized on every re-render. We could benefit from use lazy initialization. See [this](https://kentcdodds.com/blog/use-state-lazy-initialization-and-function-updates) for more info.

Great resource. Thanks for the tip. I just updated this to add lazy initialization.

> It also feels like a weird data type to live in state. Let's try to refactor and use as little state as possible. The rest that is always the same can live outside of the state.

We double-checked the code and we were able to decouple some props that are not necessarily needed on the state.
* Moved `label` to a constant and use it directly when rendering the radio button.
* Handled `checked` and `isCurrentPlan` on the radio control.